channel.go

```diff
@@ -74,8 +97,7 @@ func (c *channelPool) Put(conn GenericConn) error {
 	// block and the default case will be executed.
 	select {
```

Comment:
It looks like this will always block now since the default case has been removed. If this is intentional you should update the comment - it's a bit confusing right now.